core/src/main/java/org/infinispan/configuration/cache/IndexingConfigurationBuilder.java

```diff
@@ -320,9 +320,9 @@ public IndexingConfiguration create() {
       // check for presence of index providers that are not persistent upon restart
       boolean isVolatile = typedProperties.entrySet().stream()
                                      .anyMatch(e -> {
-                                        if (((String) e.getKey()).endsWith(DIRECTORY_PROVIDER_SUFFIX)) {
+                                        if (((String) e.getKey()).endsWith("directory.type")) {
```

Comment:
DIRECTORY_PROVIDER_SUFFIX should probably become "directory.type". It is used in other places in this class, which I think are broken now unless we make the change